Change Elasticseach support and upgrade process

jraddaoui opened the pull request:
Connects to https://github.com/artefactual/archivematica/issues/1171.
Connects to https://github.com/artefactual/archivematica/issues/1173.

scollazo commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,25 +225,43 @@ Upgrade on CentOS/Red Hat
 Storage Service.
 
-This process has two steps:
+If you have access to those locations, the recommended method for the upgrade is
```
We should include here something like "and the space used is less than 1 terabyte" or something like that, so the reindex method is not used when big aipstores are in place.

jraddaoui replied:
Good point @scollazo!

After the discussion about changing the maintenance section I have moved the command details to that page and linked them from here. I have added notices in both places about the duration of those commands for big storage locations, but I think we should avoid adding an specific storage size.

scollazo commented:
Also, I think we should outline how to set up a temporary es server in the same machine, without using external servers.

The process would be:
 - Update am using ansible or packages, but don't upgrade elasticsearch
 - Download and run a temporary ES 6.6 instance
 - Migrate from the old ES 1.7.5 to the temporary ES 6.6 instance
 - Remove ES 1.7.5
 - Install ES 6.6 using the command line instructions, or the ansible playbook
 - Copy data from temporary ES6 to the new installed ES6
 - Start services

I think we should merge this, and create a different pr to address this process

jraddaoui commented:
> Also, I think we should outline how to set up a temporary es server in the same machine, without using external servers.
> 
> The process would be:
> 
> * Update am using ansible or packages, but don't upgrade elasticsearch
> * Download and run a temporary ES 6.6 instance
> * Migrate from the old ES 1.7.5 to the temporary ES 6.6 instance
> * Remove ES 1.7.5
> * Install ES 6.6 using the command line instructions, or the ansible playbook
> * Copy data from temporary ES6 to the new installed ES6
> * Start services
> 
> I think we should merge this, and create a different pr to address this process

That would be great, thanks @scollazo!

As discussed in the chat, it's probably better to run ES 1.7 as the temporary instance using the backup data after a full upgrade.

jraddaoui commented:
@scollazo and @sallain, this is ready for review now.

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
-backlog and AIP storage locations. Respectively, these usually are:
+To complete this upgrade is important to know if you have access to your
+transfer backlog and AIP storage locations in the local filesystem.
+Respectively, these usually are located in the following paths:
```
```suggestion
These are usually located in the following paths:
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
 
-To complete these instructions you need to know the paths of your transfer
-backlog and AIP storage locations. Respectively, these usually are:
+To complete this upgrade is important to know if you have access to your
```
```suggestion
To complete this upgrade it is important to know if you have access to your
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
 * :file:`/var/archivematica/sharedDirectory/www/AIPsStore`
 
-You can confirm the paths of your installation in the Location tab of the
+You should confirm the paths of your installation in the Location tab of the
```
```suggestion
You should confirm the paths of your installation in the Locations tab of the
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
+settings and populated from the files and database information. This allows you
+to upgrade the Elasticsearch instance used in the previous version to 6.x
+without taking care of the 1.x indexes data. This process requires to execute
+two commands:
```
Multi-line, so I can't use the suggestion function, but I think you can change this paragraph a bit, for clarity:

```
Using this method, the indexes will be recreated with the new mappings and
settings and will be populated from the files and database information. This will allow you
to upgrade the Elasticsearch instance to 6.x
without having to manage the 1.x index's data. Run the following commands :
```

(I may not have the particulars 100% right but I think I got the general idea...)

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
-
-   .. code:: bash
+.. note::
+   Please notice, the execution of this commands may take a long time for big
```
```suggestion
   Please notice, the execution of this command may take a long time for big
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
-   .. code:: bash
+.. note::
+   Please notice, the execution of this commands may take a long time for big
+   AIP and Transfer Backlog storage locations, specially if the AIPs are stored
```
```suggestion
   AIP and Transfer Backlog storage locations, especially if the AIPs are stored
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
+.. note::
+   Please notice, the execution of this commands may take a long time for big
+   AIP and Transfer Backlog storage locations, specially if the AIPs are stored
+   compressed. If that's your case, you may want to give a try to the
```
```suggestion
   compressed. If that is your case, you may want to try the
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
+   Please notice, the execution of this commands may take a long time for big
+   AIP and Transfer Backlog storage locations, specially if the AIPs are stored
+   compressed. If that's your case, you may want to give a try to the
+   :ref:`reindex from another cluster method <cluster-reindex>` bellow.
```
```suggestion
   :ref:`reindex from another cluster method <cluster-reindex>`, below.
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
+
+   #. External access must be enabled in the ES instances if they are not in the
+      same machine as Archivematica.
+   #. The command accepts basic auth. parameters to connect to the ES 1.x
```
```suggestion
   #. The command accepts basic authentication parameters to connect to the ES 1.x
```

sallain commented on admin-manual/installation-setup/upgrading/upgrading.rst:
```diff
@@ -220,72 +225,99 @@ Upgrade on CentOS/Red Hat
+#. The ES 1.x host has to be white-listed in the ES 6.x "elasticsearch.yaml"
+   configuration file (e.g.: reindex.remote.whitelist: "host:9200").
+#. The command requires the ES 1.x instance URL (including protocol and port)
+   as the first argument, two optional parameters for basic auth. and other two
```
```suggestion
   as the first argument, two optional parameters for basic authentication and two other
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-
-   q = pyes.TermQuery("METS.amdSec.ns0:amdSec_list.@ID", "amdsec_8")
+Since version 0.9, Archivematica uses Elasticsearch as its search engine. It
+stores information for the AIPs and the Transers in backlog. This data can be
```
```suggestion
stores information about AIPs and Transfers in backlog. This data can be
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-   q = pyes.TermQuery("METS.amdSec.ns0:amdSec_list.@ID", "amdsec_8")
+Since version 0.9, Archivematica uses Elasticsearch as its search engine. It
+stores information for the AIPs and the Transers in backlog. This data can be
+searched from the Archival Storage and the Backlog areas of the dashboard.
```
```suggestion
searched from the Archival Storage and the Backlog tabs on the Archivematica dashboard.
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-   conn = pyes.ES('127.0.0.1:9200')
-
-   q = pyes.TermQuery("METS.amdSec.ns0:amdSec_list.@ID", "amdsec_8")
+Since version 0.9, Archivematica uses Elasticsearch as its search engine. It
```
```suggestion
Since version 0.9, Archivematica uses Elasticsearch as its search engine. Elasticsearch
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-Delete the Elasticsearch index through the GUI
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Archivematica includes two Django commands to regenerate the Elasticsearch
+indexes. This commands require access in the local filesystem to the location
```
```suggestion
indexes. These commands require access in the local filesystem to the location
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Archivematica includes two Django commands to regenerate the Elasticsearch
+indexes. This commands require access in the local filesystem to the location
+paths of the AIP and Transfer Backlog storage locations. Respectively, these
```
```suggestion
paths of the AIP and Transfer Backlog storage locations. These
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
+* :file:`/var/archivematica/sharedDirectory/www/AIPsStore/transferBacklog`
 
-Elasticsearch Head will allow you to delete an index, if need be.
+You should confirm the paths of your installation in the Location tab of the
```
```suggestion
You should confirm the paths of your installation in the Locations tab of the
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-If, for whatever reason, you need to delete an Elasticsearch index
-programmatically, this can be done with pyes using the following code.
+.. note::
+   Please notice, the execution of this command may take a long time for big
```
```suggestion
   Please note, the execution of this command may take a long time for big
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-programmatically, this can be done with pyes using the following code.
+.. note::
+   Please notice, the execution of this command may take a long time for big
+   AIP storage locations, specially if the AIPs are stored compressed.
```
```suggestion
   AIP storage locations, especially if the AIPs are stored compressed.
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
-rebuild (substituting "/your/script/location/rebuild-elasticsearch-aip-index-
-from-files" with the location of the script):
+.. note::
+   Please notice, the execution of this command may take a long time for big
```
```suggestion
   Please note, the execution of this command may take a long time for big
```

sallain commented on admin-manual/maintenance/maintenance.rst:
```diff
@@ -11,251 +11,126 @@ installation.
+* `Dejavu`_
 
-   `Elasticsearch troubleshooting`_ help from AtoM documentation.
+The indexes names are: `aips`, `aipfiles`, `transfers` and `transferfiles`.
```
```suggestion
The index names are: `aips`, `aipfiles`, `transfers` and `transferfiles`.
```

sallain reviewed (approved):
A few changes, but all small! Will approve, since I'll be at a conference for the next few days and not reliably available.

scollazo commented:
Added the procedure to update es and migrate the data on the wiki: https://wiki.archivematica.org/Update_ElasticSearch

jraddaoui committed:
Change Elasticseach support and upgrade process

Squashed commit of the following:

commit 76e65115461055d77126f805ec03c3e1b1632ff2
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Tue Feb 19 19:46:53 2019 +0100

    Add note and link to Wiki

commit 982076fb882901860c1dcd4daa016933a8cd57b5
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Tue Feb 19 19:41:58 2019 +0100

    Sara's suggestion 20

commit caeab602f4ca98d8085b37806f3ebed31d04dba2
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:30:52 2019 +0100

    Sara's suggestion 19

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 4435c992da3cc5c6f74a659de4cd18ca3d39c24f
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:30:25 2019 +0100

    Sara's suggestion 18

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit a8cc1b23bb1a818b6b5587e0dbc1184ece1c505d
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:30:11 2019 +0100

    Sara's suggestion 17

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 7344b05f39a511ee000ada378765589a3a87bd57
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:29:57 2019 +0100

    Sara's suggestion 16

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 169235efbe8ed392232ea5cfb3f267c9273bc35e
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:29:42 2019 +0100

    Sara's suggestion 15

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit d65343fce93d0d0f3f402d41f20a483cfb95f740
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:29:01 2019 +0100

    Sara's suggestion 14

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 1be39c7d22ea4f6ec0cbe1a6939817e5f472fba9
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:28:28 2019 +0100

    Sara's suggestion 13

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 969de30d632d62865896480f4d5e28daa07d939f
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:27:37 2019 +0100

    Sara's suggestion 12

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 1fc15819d6e1dcb019811152895e50346479b0ed
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:27:05 2019 +0100

    Sara's suggestion 11

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit a8774b84d2bf32c001f7a3ff7d965262af740870
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:25:07 2019 +0100

    Sara's suggestion 10

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit eb769b095a77504c8c99ba9ab65af367c58ee0fb
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:24:35 2019 +0100

    Sara's suggestion 9

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit f56adf98711eb406c120c8f842eeeaab081a8f86
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:23:54 2019 +0100

    Sara's suggestion 8

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit b76c04afcd3f8a6a355307252b4fd1fa36299bea
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:23:18 2019 +0100

    Sara's suggestion 7

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit b8d3fc2e549a968b6d4de7ddb867812407f5e628
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:22:43 2019 +0100

    Sara's suggestion 6

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 1cadffceede1cbdfeb2f8f017f9df5c3daf41c1d
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:22:11 2019 +0100

    Sara's suggestion 5

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 6d86c0dc6ec3c3e6ac4bbb86b0551958aae7b031
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:21:29 2019 +0100

    Sara's suggestion 4

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit 1e713153e46d93976d6fe19742c841795f765b11
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:19:34 2019 +0100

    Sara's suggestion 3

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit fcfec8122151e6c94159f61d5a315f57bc683b47
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:17:19 2019 +0100

    Sara's suggestion 2

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit c9af606f6a11c8d85b7c342a6cee31b3ba162a7f
Author: Sara Allain <sallain@users.noreply.github.com>
Date:   Tue Feb 19 19:17:06 2019 +0100

    Sara's suggestion 1

    Co-Authored-By: jraddaoui <raddaouimarin@gmail.com>

commit ab06f65011fc9f50e10f6d485ca0978a8426a8a6
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Fri Feb 15 17:49:46 2019 +0100

    Change Elasticseach support

    - Upgrade version support.
    - Install and upgrade processes.
    - Maintenance section.

jraddaoui commented:
Thanks @sallain and @scollazo!
